docs/scenarios_authoring/DesignerTipsAndTricks.md

```diff
@@ -22,3 +16,118 @@ While Designer GUI in most cases is self-explanatory and forgiving, there are a
 * **Deleting a scenario** - scenarios can be archived and unarchived, they are never completely deleted. 
 * **Inserting a node into the flow** - you can drop a node on the edge connecting nodes and Designer will fit it into the flow.
+
+
+## SpEL
+
+### Common errors and misunderstandings
+
+- Unlike JSON, SpEL uses curly brackets to denote the start and end of a list, e.g.:  `{1, 2, 3, 4}`. To make things more confusing, in the code hints and in the debugger Nussknacker shows lists using JSON notation for lists. So, stay alert.
+- `{:}` stands for the empty record (map). `{}` denotes an empty list.  
+
+### Handling data types problems
+
+Sometimes the Nussknacker’s data typing subsystem on which hints and validations are built is too restrictive and complains about a data type mismatch. In such a case you can use `#CONV.toAny()` helper function to trick Nussknacker’s data typing system into thinking that the data type of the entered expression is valid. You can see an example of how to use `#CONV.toAny` in the next section. 
+Note, that the `#CONV.toAny()` is pretty brutal in its working - it effectively disables all the type checking during the scenario authoring. Unless you [monitor logs and scenario metrics](./../operations_guide/Common.md#handling-typical-scenario-errors), you may not notice runtime validation errors concealed during scenario authoring by using `#CONV.toAny()`. 
+
+### Handling enums
+Both AVRO and JSON schema specs allow you to define an enumeration value, defined as a constant list of values (strings) this enum can hold. For example, let’s consider this AVRO DeliveryStatus type:
+
+```
+{
+  "type": "enum",
+  "name": "status",
+  "symbols" : ["ORDERED", "READY", "DELIVERED"]
+}
+```
+Suppose we want to define some kind of logic dependent on delivery status. If we just use this field in filter/choice we can discover that it is of type EnumSymbol and cannot be just compared with plain string with one of the allowed values:
+
+![img](img/enumComparisonToStringProblem.png)
+
+
+What the heck is EnumSymbol? It is a generic Java type representing the enum. Currently, the workaround is to convert it to string before the comparison is done, as on the screenshot below. 
+
+![img](img/enumComparisonToStringSolution.png)
+
+Now, imagine that we got one of the above enum values in a variable of type string. If you attempt to make the comparison, the Nussknacker typing system will complain:
+
+![img](img/enumComparisonWithVariable.png)
+
+The workaround here is to use the #CONV.toAny() helper to trick the typing subsystem into thinking that it is a valid enum symbol:
+
+![img](img/enumToAnyConversion.png)
+
+### Non-trivial operations on JSON records
+
+Map operator `.!` is truly powerful; one can achieve a lot of magic with it. A couple of examples.
```

Comment:
We link to Spring docs from the introduction on SpEL. I looked at it and unless you know and enjoy Java wordy syntax, you would be intimidated rather than informed by the context of the Spring docs. So I am against.

Reply:
I think "map" comes from Python. Anyway, believing that this would overwhelm the readers,  I removed mentions of "to map" and did not want to use "collection projection" instead. I use neutral "transforming lists". If you know spreadsheets, but never used Python, not to mention Java, then the project / map operator is a surprise. Hence the example in Tips & Tricks. Without using its official name.